tests/http-proxy.rs

```diff
@@ -0,0 +1,52 @@
+extern crate hyper;
+extern crate alacrity;
+
+use hyper::server::{Server, Request, Response, Handler};
+use hyper::Client;
+use std::io::Read;
+use std::net::SocketAddr;
+
+
+fn with_server<H: Handler + 'static, R>(handle: H, test: &Fn(u16) -> R) -> R {
+    let mut server = Server::http("localhost:0").unwrap().handle(handle).unwrap();
+    let port = server.socket.port();
+
+    // test directly against http server
+    let result_direct = test(port);
+
+    // test against proxy
+    let proxy_addr = "127.0.0.1:8081".to_string();
+    let addr = proxy_addr.parse::<SocketAddr>().unwrap();
+//    alacrity::new_proxy(addr, socket_addr(port));
+//    let result_proxy = test(8081);
```

Comment:
For the first point, I think that hyper is already taking care of that.
When I call `Server::http("localhost:0").unwrap().handle(handle).unwrap()`, the main thread is not blocked.